lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/driver/evaluator/EvaluatorManager.java

```diff
@@ -276,7 +278,16 @@ public void onNext(final Alarm alarm) {
    */
   public boolean isClosed() {
     return this.messageDispatcher.isEmpty() &&
-           this.stateManager.isDoneOrFailedOrKilled();
+        (this.stateManager.isDoneOrFailedOrKilled());
+  }
+
+  /**
+   * Return true if the state is DONE, FAILED, KILLED, or CLOSING,
+   * <em>and</em> there are no messages queued or in processing.
+   */
+  public boolean isClosedOrClosing() {
+    return this.messageDispatcher.isEmpty() &&
```

Comment:
This function is called in `Evaluators.close()` to not call `EvaluatorManager.close()` if the evaluator is already closed or in closing state.
Would it be not needed?